### ISSUE TITLE
Extend what the Location / Camera / Microphone toggles provide to Sensors

### ISSUE THREAD

**Issue by rhclayto** [GrapheneOS/os-issue-tracker#890]
Following on from https://github.com/GrapheneOS/os-issue-tracker/issues/806#issuecomment-1003208987 , I am requesting additional toggles for other sensors besides Location/Camera/Microphone. In the comment linked above, @thestinger wrote 'It would be entirely possible / sensible to extend what the Location / Camera / Microphone toggles provide to Sensors but that's not what is being asked of us.' So I am asking.

Best of all from a user perspective would be a universal 'sensors off' toggle to completely disable all sensors with a single user interaction, rather than flipping a series of toggles, but designed in 'a superior, more compatible way' that doesn't use Developer Options.

Out of curiosity, what is technically wrong or problematic with using `SensorPrivacyService` to implement a 'sensors off' feature?

**Comment by novocaiin** [comment]
Maybe also have the ability to auto-disable location, camera, microphone etc. after they haven't been used for a period of time.

Say, I make a phone call, it requests to re-enable the mic.
After I'm done with the call, instead of me having to do it manually, it could turn off on its own after 5 or so minutes of inactivity.

**Comment by rhclayto** [comment]
I like that idea too.

With regard to a master switch for sensors, it would be analogous to the Airplane Mode toggle, where you have individual toggles for data & WiFi which are both controlled by Airplane Mode. Something similar for other sensors would be super useful. Toggle Airplane More & Sensors Off & you have a comatose phone in two quick steps. Then selectively turn individual sensors on as needed. And/or configurable auto on/off.

**Comment by gordonel** [comment]
There's already a master switch for sensors in every Android since at least 9 or 10. To activate it, go to developer options > Quick settings developer tiles > Sensors Off

**Comment by thestinger** [comment]
No, that's not the same thing, and is meant for developers.

**Comment by thestinger** [comment]
That disables the camera, microphone and other sensors too. It doesn't disable them in the same way and is a developer feature. The camera and microphone toggles are friendlier to apps and more robust.

**Comment by gordonel** [comment]
> That disables the camera, microphone and other sensors too. It doesn't disable them in the same way and is a developer feature. The camera and microphone toggles are friendlier to apps and more robust.

I kinda agree. Mic and Camera toggles are more useful and would actually disable the mic for dialer apps, for instance. Still, it is a useful feature. Yes, it can sometimes break apps, but it's a small price to pay

**Comment by rhclayto** [comment]
> There's already a master switch for sensors in every Android since at least 9 or 10. To activate it, go to developer options > Quick settings developer tiles > Sensors Off. It will then appear in the notifications drawer

This feature request is to make a similar feature done the GrapheneOS way. I have no problem using the current 'Sensors Off' toggle in the developer options, & I use it every day. It isn't available in user profiles however, which is why I originally requested this feature be added to GrapheneOS https://github.com/GrapheneOS/os-issue-tracker/issues/806#issuecomment-1003208987 . Daniel Mickay, lead developer here (thestinger), is opposed to using the `SensorPrivacyService` built in to Android (which is what the Sensors Off toggle uses). The Android docs do state that the service is intended both for developers as well as privacy-conscious users, & that the current Sensors Off toggle is placed in the developer options menu because it is a reference implementation of the service, which can be adopted by OS & device manufacturers as a user-facing toggle. That's what I originally hoped for, but the idea was shot down. So I opened this issue for the reasons stated in the issue above. I still would love a universal Sensors Off toggle, however implemented, as well as individual toggles for Camera/ Mic/Location as already exist. I'll take what I can get, though. I tried very cursorily to make an Android app that would do this, but my Android skills are rudimentary at best.
